docs/DesignDocs/CustomAttestation.md

```diff
@@ -0,0 +1,747 @@
+  - The `claims` field contains key-value pairs that can be verified by the
+    caller. This will have the similar contents as the `oe_identity_t` field
+    in the `oe_report_t` struct returned by `oe_verify_report` and any custom
+    claims that were passed to the `get_evidence` function.
+  - The `input_validation_time` is needed to validate the evidence against a
+    specific time.
+
+###  Known Open Enclave Claims
+
+- Each plugin's `verify_evidence` function must, at minimum, return the
+  following claims (mapped from the `oe_identity_t`):
+  
+| Claim Name       | Claim Value Type   | Description                                                          |
+|:-----------------|:-------------------|:---------------------------------------------------------------------|
+| id_version       | uint32_t           | Claims version. Must be 0                                            |
+| security_version | uint32_t           | Security version of the enclave. (ISVN for SGX).                     |
+| attributes       | uint64_t           | Attributes flags for the evidence: <br/> `OE_REPORT_ATTRIBUTES_DEBUG`: The evidence is for a debug enclave.<br/> `OE_REPORT_ATTRIBUTES_REMOTE`: The evidence can be used for remote attestation.   |
+| unique_id        | uint8_t[32]        | The unique ID for the enclave (MRENCLAVE for SGX).                   |
+| signer_id        | uint8_t[32]        | The signer ID for the enclave (MRSIGNER for SGX).                    |
+| product_id       | uint8_t[32]        | The product ID for the enclave (ISVPRODID for SGX).                  |
+| validity_from    | oe_datetime_t      | Overall datetime from which the evidence and endorsements are valid. |
+| validity_until   | oe_datetime_t      | Overall datetime at which the evidence and endorsements expire.      |
+
+### Built-in SGX Plugin
+
+The current Open Enclave attestation only works on SGX platforms, so it will
+be moved to an SGX plugin. Most of the current Open Enclave APIs can be mapped
+directly to the plugin APIs. For the `on_register` and `on_unregister`  APIs,
+they can simply be no-ops. `oe_get_report` can be mapped to the `get_evidence` API and
+`oe_verify_report` can be mapped to the `verify_evidence` API.
+
+### SGX Plug-In Definitions
+
+`sgx_attestation_plugin.h`
+```C
+
+/* Define the uuid. */
+#define SGX_PLUGIN_UUID                 \
+{                                       \
+ 0x2f, 0x50, 0xdc, 0xb4,                \
+ 0x79, 0x9c,                            \
+ 0x45, 0x07,                            \
+ 0xa1, 0xe9,                            \
+ 0x86, 0x2c, 0x62, 0x9b, 0x76, 0x2a}    \
+}
+
+/*! sgx_attestation_plugin
+ * 
+ * Return the SGX attesation plug-in.
+ */
+oe_attestation_plugin_t* sgx_attestation_plugin();
+
+```
+
+`sgx_attestation_plugin.c`
+```C
+#include "sgx_attestation_plugin.h"
+
+/* Struct containing params for oe report functions. */
+struct sgx_params_t {
```

Comment:
Since the flags field is almost like a required input parameter that needs be specified every time for the oe_get_evidence() function, perhaps it may be better to expose it as a separate parameter like oe_get_report()?

Reply:
Added a flags parameter